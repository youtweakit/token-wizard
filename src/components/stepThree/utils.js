import moment from 'moment';

export function defaultCompanyStartDate() {
  let crowdsaleStartDate = moment().add(5, 'minutes');
  let crowdsaleStartDateFormatted = crowdsaleStartDate.format('YYYY-MM-DDTHH:mm');
  return crowdsaleStartDateFormatted;
}

export const defaultCompanyEndDate = (startDate) => {
  const crowdsaleEndDate = moment(startDate).add(4, 'days').startOf('day')
  return crowdsaleEndDate.format('YYYY-MM-DDTHH:mm')
}
