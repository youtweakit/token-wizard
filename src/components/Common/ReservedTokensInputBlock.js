import React, { Component } from 'react'
import Web3 from 'web3'
import '../../assets/stylesheets/application.css'
import { InputField } from './InputField'
import { RadioInputField } from './RadioInputField'
import { TEXT_FIELDS, VALIDATION_TYPES } from '../../utils/constants'
import update from 'immutability-helper'
import ReservedTokensItem from './ReservedTokensItem'
import { observer } from 'mobx-react';
import { NumericInput } from './NumericInput'

const { VALID, INVALID } = VALIDATION_TYPES
const { ADDRESS, DIMENSION, VALUE } = TEXT_FIELDS

@observer
export class ReservedTokensInputBlock extends Component {
  constructor (props) {
    super(props)

    this.state = {
      addr: '',
      dim: 'tokens',
      val: '',
      validation: {
        address: {
          pristine: true,
          valid: INVALID
        },
        value: {
          pristine: true,
          valid: INVALID
        }
      }
    }
  }

  clearInput () {
    this.setState({
      addr: '',
      dim: 'tokens',
      val: ''
    })
  }

  updateReservedTokenInput = (value, property) => {
    this.setState({
      [property]: value
    })
  }

  addReservedTokensItem = () => {
    const { addr, dim, val } = this.state

    this.setState(update(this.state, {
      validation: {
        address: {
          pristine: { $set: false }
        },
        value: {
          pristine: { $set: false }
        }
      }
    }))

    const validFields = this.state.validation.address.valid === VALID && this.state.validation.value.valid === VALID

    if (!addr || !dim || !val || !validFields) {
      return
    }

    this.setState(update(this.state, {
      validation: {
        address: {
          $set: {
            pristine: true,
            valid: INVALID
          }
        },
        value: {
          $set: {
            pristine: true,
            valid: INVALID
          }
        }
      }
    }))

    this.clearInput()

    let newToken = {
      addr: addr,
      dim: dim,
      val: val
    }

    this.props.addReservedTokensItem(newToken)
  }

  handleAddressChange = address => {
    const isAddressValid = Web3.utils.isAddress(address) ? VALID : INVALID

    const newState = update(this.state, {
      validation: {
        address: {
          $set: {
            pristine: false,
            valid: isAddressValid
          },
        },
      },
    })
    newState.addr = address

    this.setState(newState)
  }

  handleValueChange = ({ value, pristine, valid }) => {
    const newState = update(this.state, {
      validation: {
        value: {
          $set: {
            pristine: pristine,
            valid: valid
          }
        }
      }
    })
    newState.val = value

    this.setState(newState)
  }

  render () {
    const reservedTokensElements = this.props.tokens.map((token, index) => {
      return (
        <ReservedTokensItem
          key={index.toString()}
          num={index}
          addr={token.addr}
          dim={token.dim}
          val={token.val}
          onRemove={index => this.props.removeReservedToken(index)}
        />
      )
    })

    let valueInputParams = null

    if (this.state.dim === 'tokens') {
      valueInputParams = {
        min: !this.props.decimals ? 0 : Number(`1e-${this.props.decimals}`),
        maxDecimals: !this.props.decimals ? 0 : this.props.decimals,
        errorMessage: 'Value must be positive and decimals should not exceed the amount of decimals specified',
        description: 'Value in tokens. Don\'t forget to click + button for each reserved token.'
      }
    } else if (this.state.dim === 'percentage') {
      valueInputParams = {
        min: Number.MIN_VALUE,
        errorMessage: 'Value must be positive',
        description: 'Value in percentage. Don\'t forget to click + button for each reserved token.'
      }
    } else {
      console.error(`unrecognized dimension '${this.state.dim}'`)
    }

    return (
      <div className="reserved-tokens-container">
        <div className="reserved-tokens-input-container">
          <div className="reserved-tokens-input-container-inner">
            <InputField
              side="reserved-tokens-input-property reserved-tokens-input-property-left"
              type="text"
              title={ADDRESS}
              value={this.state.addr}
              onChange={e => this.handleAddressChange(e.target.value)}
              description="Address where to send reserved tokens."
              pristine={this.state.validation.address.pristine}
              valid={this.state.validation.address.valid}
              errorMessage="The inserted address is invalid"
            />
            <RadioInputField
              extraClassName="reserved-tokens-input-property reserved-tokens-input-property-middle"
              title={DIMENSION}
              items={[{ label: 'tokens', value: 'tokens' }, { label: 'percentage', value: 'percentage' }]}
              selectedItem={this.state.dim}
              onChange={e => this.updateReservedTokenInput(e.target.value, 'dim')}
              description="Fixed amount or % of crowdsaled tokens. Will be deposited to the account after finalization
               of the crowdsale."
            />
            <NumericInput
              side="reserved-tokens-input-property reserved-tokens-input-property-right"
              title={VALUE}
              value={this.state.val}
              pristine={this.state.validation.value.pristine}
              valid={this.state.validation.value.valid}
              acceptFloat={true}
              onValueUpdate={this.handleValueChange}
              {...valueInputParams}
            />
          </div>
          <div className="plus-button-container">
            <div onClick={e => this.addReservedTokensItem()} className="button button_fill button_fill_plus"/>
          </div>
        </div>
        {reservedTokensElements}
      </div>
    )
  }
}
