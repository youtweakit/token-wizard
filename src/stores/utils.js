import { setFlatFileContentToState, toFixed } from '../utils/utils'
import { contractStore, tokenStore, tierStore, web3Store } from './index'
import { BigNumber } from 'bignumber.js'

export function getWhiteListWithCapCrowdsaleAssets() {
  const contractsRoute = './contracts/'
  const crowdsaleFilename = 'CrowdsaleWhiteListWithCap'
  const binAbi = ['bin', 'abi']

  const crowdsaleFiles = ['sol', ...binAbi].map(ext => `${contractsRoute}${crowdsaleFilename}_flat.${ext}`)
  const tokenFiles = binAbi.map(ext => `${contractsRoute}${crowdsaleFilename}Token_flat.${ext}`)
  const pricingFiles = binAbi.map(ext => `${contractsRoute}${crowdsaleFilename}PricingStrategy_flat.${ext}`)
  const finalizeFiles = binAbi.map(ext => `${contractsRoute}FinalizeAgent_flat.${ext}`)
  const nullFiles = binAbi.map(ext => `${contractsRoute}NullFinalizeAgent_flat.${ext}`)
  const registryFiles = binAbi.map(ext => `${contractsRoute}Registry_flat.${ext}`)

  const states = crowdsaleFiles.concat(tokenFiles, pricingFiles, finalizeFiles, nullFiles, registryFiles)
    .map(setFlatFileContentToState)

  return Promise.all(states)
    .then(state => {
      contractStore.setContractProperty('crowdsale', 'src', state[0])
      contractStore.setContractProperty('crowdsale', 'bin', state[1])
      contractStore.setContractProperty('crowdsale', 'abi', JSON.parse(state[2]))
      contractStore.setContractProperty('token', 'bin', state[3])
      contractStore.setContractProperty('token', 'abi', JSON.parse(state[4]))
      contractStore.setContractProperty('pricingStrategy', 'bin', state[5])
      contractStore.setContractProperty('pricingStrategy', 'abi', JSON.parse(state[6]))
      contractStore.setContractProperty('finalizeAgent', 'bin', state[7])
      contractStore.setContractProperty('finalizeAgent', 'abi', JSON.parse(state[8]))
      contractStore.setContractProperty('nullFinalizeAgent', 'bin', state[9])
      contractStore.setContractProperty('nullFinalizeAgent', 'abi', JSON.parse(state[10]))
      contractStore.setContractProperty('registry', 'bin', state[11])
      contractStore.setContractProperty('registry', 'abi', JSON.parse(state[12]))
      return contractStore
    })
}

export const getconstructorParams = (abiConstructor, vals, crowdsaleNum, isCrowdsale) => {
  let params = {"types": [], "vals": []};
  if (!abiConstructor) return params;

  for (let j = 0; j < abiConstructor.length; j++) {
    let inp = abiConstructor[j];
    params.types.push(inp.type);
    if (vals.length > 0) {
      params.vals.push(vals[j]);
    } else {
      switch(inp.name) {
        case "_start":
          params.vals.push(toFixed(new Date(tierStore.tiers[crowdsaleNum].startTime).getTime() / 1000).toString());
          break;
        case "_end":
          params.vals.push(toFixed(new Date(tierStore.tiers[crowdsaleNum].endTime).getTime() / 1000).toString());
          break;
        case "_rate":
          params.vals.push(tierStore.tiers[crowdsaleNum].rate);
          break;
        case "_multisigWallet":
          //params.vals.push(contractStore.multisig.addr);
          params.vals.push(tierStore.tiers[0].walletAddress);
          break;
        case "_pricingStrategy":
          params.vals.push(contractStore.pricingStrategy.addr[crowdsaleNum]);
          break;
        case "_token":
          params.vals.push(contractStore.token.addr);
          break;
        case "_crowdsale":
          params.vals.push(contractStore.crowdsale.addr[crowdsaleNum]);
          break;
        case "_name":
          if (isCrowdsale) {
            params.vals.push(tierStore.tiers[crowdsaleNum].tier);
          } else {
            params.vals.push(tokenStore.name);
          }
          break;
        case "_symbol":
          params.vals.push(tokenStore.ticker);
          break;
        case "_decimals":
          params.vals.push(tokenStore.decimals);
          break;
        case "_globalMinCap":
          params.vals.push(tierStore.tiers[0].whitelistEnabled !== 'yes' ? tokenStore.globalmincap ? toFixed(tokenStore.globalmincap * 10 ** tokenStore.decimals).toString() : 0 : 0)
          break;
        case "_initialSupply":
          params.vals.push(tokenStore.supply);
          break;
        case "_maximumSellableTokens":
          params.vals.push(toFixed(tierStore.tiers[crowdsaleNum].supply * 10**tokenStore.decimals).toString());
          break;
        case "_minimumFundingGoal":
          params.vals.push(0);
          break;
        case "_mintable":
          params.vals.push(true);
          break;
        case "_oneTokenInWei":
          BigNumber.config({ DECIMAL_PLACES: 18 })
          const rate = new BigNumber(tierStore.tiers[crowdsaleNum].rate)
          const tokenInEther = rate.pow(-1).toFixed()
          params.vals.push(web3Store.web3.utils.toWei(tokenInEther, "ether"))
          break;
        case "_isUpdatable":
          params.vals.push(tierStore.tiers[crowdsaleNum].updatable ? tierStore.tiers[crowdsaleNum].updatable==="on" ? true : false : false);
          break;
        case "_isWhiteListed":
          params.vals.push(tierStore.tiers[0].whitelistEnabled ? tierStore.tiers[0].whitelistEnabled === "yes" : false)
          break;
        default:
          params.vals.push("");
          break;
      }
    }
  }
  return params;
}
