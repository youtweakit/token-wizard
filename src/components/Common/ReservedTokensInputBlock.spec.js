import React from 'react'
import { ReservedTokensInputBlock } from './ReservedTokensInputBlock'
import { VALIDATION_TYPES } from '../../utils/constants'
import renderer from 'react-test-renderer'
import Adapter from 'enzyme-adapter-react-15'
import { configure, mount } from 'enzyme'

const { VALID, INVALID } = VALIDATION_TYPES

configure({ adapter: new Adapter() })

describe('ReservedTokensInputBlock', () => {
  let tokenList
  let addCallback
  let removeCallback
  let decimals
  let wrapperMemo, wrapper
  let addressInputMemo, addressInput
  let valueInputMemo, valueInput
  let addressStateMemo, addressState
  let valueStateMemo, valueState

  beforeEach(() => {
    tokenList = [
      {
        addr: '0x90f8bf6a479f320ead074411a4b0e7944ea8c9c1',
        dim: 'tokens',
        value: '120'
      },
      {
        addr: '0x90f8bf6a479f320ead074411a4b0e7944ea8c9c1',
        dim: 'percentage',
        value: '105'
      },
      {
        addr: '0xffcf8fdee72ac11b5c542428b35eef5769c409f0',
        dim: 'percentage',
        value: '100'
      },
      {
        addr: '0x22d491bde2303f2f43325b2108d26f1eaba1e32b',
        dim: 'tokens',
        value: '20'
      }
    ]
    addCallback = jest.fn()
    removeCallback = jest.fn()
    decimals = 3

    wrapperMemo = undefined
    wrapper = () => wrapperMemo || (wrapperMemo = mount(
      <ReservedTokensInputBlock
        tokens={tokenList}
        decimals={decimals}
        addReservedTokensItem={addCallback}
        removeReservedToken={removeCallback}
      />
    ))

    addressInputMemo = undefined
    addressInput = () => addressInputMemo || (addressInputMemo = wrapper().find('input[type="text"]').at(0))

    valueInputMemo = undefined
    valueInput = () => valueInputMemo || (valueInputMemo = wrapper().find('input[type="number"]').at(0))

    addressStateMemo = undefined
    addressState = () => addressStateMemo || (addressStateMemo = wrapper().state('validation').address)

    valueStateMemo = undefined
    valueState = () => valueStateMemo || (valueStateMemo = wrapper().state('validation').value)
  })

  describe('render', () => {
    let componentMemo, component
    beforeEach(() => {
      componentMemo = undefined
      component = () => componentMemo || (componentMemo = renderer.create(
        <ReservedTokensInputBlock
          tokens={tokenList}
          decimals={decimals}
          addReservedTokensItem={addCallback}
          removeReservedToken={removeCallback}
        />
      ))
    })

    it('Should render the component for tokens', () => {
      expect(component().toJSON()).toMatchSnapshot('tokens')
    })

    it('Should render the component for percentage', () => {
      wrapper().find('input[type="radio"]').at(1).simulate('change')
      expect(component().toJSON()).toMatchSnapshot('percentage')
    })

    it('Should render all the tokens passed', () => {
      expect(wrapper().find('.reserved-tokens-item-container')).toHaveLength(tokenList.length)
    })
  })

  describe('Address field', () => {
    [
      { value: '0x123', expected: false },
      { value: '0x90F8bF6A479f320EAD074411A4B0E7944EA8C9C1', expected: false },
      { value: '0x90f8bf6a479f320ead074411a4b0e7944ea8c9c1', expected: true },
      { value: '0X90f8bf6a479f320ead074411a4b0e7944ea8c9c1', expected: true },
      { value: '0x90F8BF6A479F320EAD074411A4B0E7944EA8C9C1', expected: true },
      { value: '0X90F8BF6A479F320EAD074411A4B0E7944EA8C9C1', expected: true },
    ].forEach(testCase => {
      const action = testCase.expected ? 'pass' : 'fail'
      const validity = testCase.expected ? VALID : INVALID

      it(`Should ${action} for '${testCase.value}'`, () => {
        const handleAddressChange = jest.spyOn(wrapper().instance(), 'handleAddressChange')

        wrapper().update()

        addressInput().simulate('change', { target: { value: testCase.value } })
        expect(handleAddressChange).toHaveBeenCalledTimes(1)
        expect(handleAddressChange).toHaveBeenCalledWith(testCase.value)
        expect(addressState().pristine).toBeFalsy()
        expect(addressState().valid).toBe(validity)
      })
    })
  })

  describe('Dimension selection', () => {
    it('Should set "tokens" as selected dimension', () => {
      const updateReservedTokenInput = jest.spyOn(wrapper().instance(), 'updateReservedTokenInput')

      wrapper().update()

      wrapper().find('input[type="radio"]').at(0).simulate('change')
      expect(updateReservedTokenInput).toHaveBeenCalledWith('tokens', 'dim')
      expect(updateReservedTokenInput).toHaveBeenCalledTimes(1)
      expect(wrapper().state('dim')).toBe('tokens')
    })
    it('Should set "percentage" as selected dimension', () => {
      const updateReservedTokenInput = jest.spyOn(wrapper().instance(), 'updateReservedTokenInput')

      wrapper().update()

      wrapper().find('input[type="radio"]').at(1).simulate('change')
      expect(updateReservedTokenInput).toHaveBeenCalledWith('percentage', 'dim')
      expect(updateReservedTokenInput).toHaveBeenCalledTimes(1)
      expect(wrapper().state('dim')).toBe('percentage')
    })
    it('Should set "tokens" as selected dimension after selecting percentage', () => {
      const updateReservedTokenInput = jest.spyOn(wrapper().instance(), 'updateReservedTokenInput')

      wrapper().update()

      wrapper().find('input[type="radio"]').at(1).simulate('change')
      expect(updateReservedTokenInput).toHaveBeenCalledWith('percentage', 'dim')
      expect(updateReservedTokenInput).toHaveBeenCalledTimes(1)
      expect(wrapper().state('dim')).toBe('percentage')

      wrapper().find('input[type="radio"]').at(0).simulate('change')
      expect(updateReservedTokenInput).toHaveBeenCalledWith('tokens', 'dim')
      expect(updateReservedTokenInput).toHaveBeenCalledTimes(2)
      expect(wrapper().state('dim')).toBe('tokens')
    })
    it('Should fail if state.dim is not "percentage" or "tokens"', () => {
      wrapper().setState({dim: 'height'})
      const radio = wrapper().find('input[type="radio"]')
      const tokens = radio.at(0)
      const percentage = radio.at(1)

      expect(tokens.props().checked).toBeFalsy()
      expect(percentage.props().checked).toBeFalsy()
    })
  })

  describe('Value field', () => {
    describe('tokens', () => {
      [
        { value: '0', expected: false },
        { value: '-10', expected: false },
        { value: -10, expected: false },
        { value: 123.1234, expected: false },
        { value: '10', expected: true },
        { value: 10, expected: true },
        { value: '0.1', expected: true },
        { value: '1e-3', expected: true },
        { value: '1.3', expected: true },
      ].forEach(testCase => {
        const action = testCase.expected ? 'pass' : 'fail'
        const validity = testCase.expected ? VALID : INVALID
        const calledWith = {
          value: testCase.value === '' ? '' : parseFloat(testCase.value),
          pristine: false,
          valid: validity
        }

        it(`Should ${action} for '${testCase.value}' with ${decimals} decimals`, () => {
          const handleValueChange = jest.spyOn(wrapper().instance(), 'handleValueChange')

          wrapper().update()
          wrapper().find('input[type="radio"]').at(0).simulate('change')
          valueInput().simulate('change', { target: { value: testCase.value } })

          expect(handleValueChange).toHaveBeenCalledTimes(1)
          expect(handleValueChange).toHaveBeenCalledWith(calledWith)
        })
      })
    })

    describe('percentage', () => {
      [
        { value: '0', expected: false },
        { value: '-10', expected: false },
        { value: -10, expected: false },
        { value: '10', expected: true },
        { value: 10, expected: true },
        { value: '0.1', expected: true },
        { value: '1e-3', expected: true },
        { value: '1.3', expected: true },
        { value: 123.1234, expected: true },
      ].forEach(testCase => {
        const action = testCase.expected ? 'pass' : 'fail'
        const validity = testCase.expected ? VALID : INVALID
        const calledWith = {
          value: testCase.value === '' ? '' : parseFloat(testCase.value),
          pristine: false,
          valid: validity
        }

        it(`Should ${action} for '${testCase.value}'`, () => {
          const handleValueChange = jest.spyOn(wrapper().instance(), 'handleValueChange')

          wrapper().update()
          wrapper().find('input[type="radio"]').at(1).simulate('change')
          valueInput().simulate('change', { target: { value: testCase.value } })

          expect(handleValueChange).toHaveBeenCalledTimes(2)
          expect(handleValueChange).toHaveBeenCalledWith({
            value: '',
            pristine: true,
            valid: INVALID
          })
          expect(handleValueChange).toHaveBeenCalledWith(calledWith)
        })
      })
    })
  })

  describe('Callbacks', () => {
    describe('addReservedTokensItem', () => {
      [
        {
          value: {
            addr: '0x90f8bf6a479f320ead074411a4b0e7944ea8c9c1',
            dim: 'tokens',
            val: '120'
          },
          expected: true
        },
        {
          value: {
            addr: '0x90f8bf6a479f320ead074411a4b0e7944ea8c9c1',
            dim: 'tokens',
            val: '0.001'
          },
          expected: true
        },
        {
          value: {
            addr: '0x90f8bf6a479f320ead074411a4b0e7944ea8c9c1',
            dim: 'tokens',
            val: ''
          },
          expected: false
        },
        {
          value: {
            addr: '0x90f8bf6a479f320ead074411a4b0e7944ea8c9c1',
            dim: 'tokens',
            val: '0'
          },
          expected: false
        },
        {
          value: {
            addr: '0x90f8bf6a479f320ead074411a4b0e7944ea8c9c1',
            dim: 'tokens',
            val: '120.1234'
          },
          expected: false
        },
        {
          value: {
            addr: '',
            dim: 'tokens',
            val: '120'
          },
          expected: false
        },
        {
          value: {
            addr: '0x90f8bf6a4',
            dim: 'tokens',
            val: '120'
          },
          expected: false
        },
        {
          value: {
            addr: '0x90f8bf6a479f320ead074411a4b0e7944ea8c9c10x90f8bf6a479f320ead074411a4b0e7944ea8c9c1',
            dim: 'tokens',
            val: '120'
          },
          expected: false
        },
        {
          value: {
            addr: '0x90f8bf6a479f320ead074411a4b0e7944ea8c9c1',
            dim: 'percentage',
            val: '120'
          },
          expected: true
        },
        {
          value: {
            addr: '0x90f8bf6a479f320ead074411a4b0e7944ea8c9c1',
            dim: 'percentage',
            val: '120.123'
          },
          expected: true
        },
        {
          value: {
            addr: '0x90f8bf6a479f320ead074411a4b0e7944ea8c9c1',
            dim: 'percentage',
            val: ''
          },
          expected: false
        },
        {
          value: {
            addr: '',
            dim: 'percentage',
            val: '120.123'
          },
          expected: false
        },
        {
          value: {
            addr: '0x90f8bf6a479f320ead074411a4b0e7944ea8c9c1',
            dim: 'percentage',
            val: '0'
          },
          expected: false
        },
      ].forEach(testCase => {
        const radioIndex = testCase.value.dim === 'tokens' ? 0 : 1
        const action = testCase.expected ? 'call' : 'not call'
        const callTimes = testCase.expected ? 1 : 0

        it(`Should ${action} addReservedTokensItem callback for: ${JSON.stringify(testCase.value)}`, () => {
          wrapper().find('input[type="radio"]').at(radioIndex).simulate('change')

          addressInput().simulate('change', { target: { value: testCase.value.addr } })
          valueInput().simulate('change', { target: { value: testCase.value.val } })

          wrapper().find('.plus-button-container').childAt(0).simulate('click')

          const { value } = testCase
          value.val = testCase.value.val === '' ? '' : parseFloat(testCase.value.val)

          if (testCase.expect) {
            expect(addCallback).toHaveBeenCalledWith(value)
          }
          expect(addCallback).toHaveBeenCalledTimes(callTimes)
        })

      })
    })

    describe('removeReservedToken', () => {
      it('Should call removeReservedToken callback', () => {
        wrapper().find('.reserved-tokens-item-empty').children('a').at(0).simulate('click')
        expect(removeCallback).toHaveBeenCalledTimes(1)
      })
    })
  })
})
