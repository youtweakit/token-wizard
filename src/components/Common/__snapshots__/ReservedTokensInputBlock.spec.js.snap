// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`percentage 1`] = `
<div
  className="reserved-tokens-container"
>
  <div
    className="reserved-tokens-input-container"
  >
    <div
      className="reserved-tokens-input-container-inner"
    >
      <div
        className="reserved-tokens-input-property reserved-tokens-input-property-left"
      >
        <label
          className="label"
        >
          Address
        </label>
        <input
          className="input"
          disabled={undefined}
          onBlur={undefined}
          onChange={[Function]}
          onKeyPress={undefined}
          onPaste={undefined}
          style={undefined}
          type="text"
          value=""
        />
        <p
          className="description"
        >
          Address where to send reserved tokens.
        </p>
        <p
          style={
            Object {
              "color": "red",
              "fontSize": "12px",
              "fontWeight": "bold",
              "height": "10px",
              "width": "100%",
            }
          }
        />
      </div>
      <div
        className="reserved-tokens-input-property reserved-tokens-input-property-middle"
      >
        <label
          className="label"
        >
          Dimension
        </label>
        <div
          className="radios-inline"
        >
          <label
            className="radio-inline"
          >
            <input
              checked={true}
              onChange={[Function]}
              type="radio"
              value="tokens"
            />
            <span
              className="title"
            >
              tokens
            </span>
          </label>
          <label
            className="radio-inline"
          >
            <input
              checked={false}
              onChange={[Function]}
              type="radio"
              value="percentage"
            />
            <span
              className="title"
            >
              percentage
            </span>
          </label>
        </div>
        <p
          className="description"
        >
          Fixed amount or % of crowdsaled tokens. Will be deposited to the account after finalization of the crowdsale.
        </p>
      </div>
      <div
        className="reserved-tokens-input-property reserved-tokens-input-property-right"
      >
        <label
          className="label"
        >
          Value
        </label>
        <input
          className="input"
          disabled={undefined}
          onBlur={undefined}
          onChange={[Function]}
          onKeyPress={[Function]}
          onPaste={[Function]}
          style={undefined}
          type="number"
          value=""
        />
        <p
          className="description"
        >
          Value in tokens. Don't forget to click + button for each reserved token.
        </p>
        <p
          style={
            Object {
              "color": "red",
              "fontSize": "12px",
              "fontWeight": "bold",
              "height": "10px",
              "width": "100%",
            }
          }
        />
      </div>
    </div>
    <div
      className="plus-button-container"
    >
      <div
        className="button button_fill button_fill_plus"
        onClick={[Function]}
      />
    </div>
  </div>
  <div
    className="reserved-tokens-item-container"
  >
    <div
      className="reserved-tokens-item-container-inner"
    >
      <span
        className="reserved-tokens-item reserved-tokens-item-left"
      >
        0x90f8bf6a479f320ead074411a4b0e7944ea8c9c1
      </span>
      <span
        className="reserved-tokens-item reserved-tokens-item-middle"
      >
        tokens
      </span>
      <span
        className="reserved-tokens-item reserved-tokens-item-right"
      />
    </div>
    <div
      className="reserved-tokens-item-empty"
    >
      <a
        onClick={[Function]}
      >
        <span
          className="item-remove"
        />
      </a>
    </div>
  </div>
  <div
    className="reserved-tokens-item-container"
  >
    <div
      className="reserved-tokens-item-container-inner"
    >
      <span
        className="reserved-tokens-item reserved-tokens-item-left"
      >
        0x90f8bf6a479f320ead074411a4b0e7944ea8c9c1
      </span>
      <span
        className="reserved-tokens-item reserved-tokens-item-middle"
      >
        percentage
      </span>
      <span
        className="reserved-tokens-item reserved-tokens-item-right"
      />
    </div>
    <div
      className="reserved-tokens-item-empty"
    >
      <a
        onClick={[Function]}
      >
        <span
          className="item-remove"
        />
      </a>
    </div>
  </div>
  <div
    className="reserved-tokens-item-container"
  >
    <div
      className="reserved-tokens-item-container-inner"
    >
      <span
        className="reserved-tokens-item reserved-tokens-item-left"
      >
        0xffcf8fdee72ac11b5c542428b35eef5769c409f0
      </span>
      <span
        className="reserved-tokens-item reserved-tokens-item-middle"
      >
        percentage
      </span>
      <span
        className="reserved-tokens-item reserved-tokens-item-right"
      />
    </div>
    <div
      className="reserved-tokens-item-empty"
    >
      <a
        onClick={[Function]}
      >
        <span
          className="item-remove"
        />
      </a>
    </div>
  </div>
  <div
    className="reserved-tokens-item-container"
  >
    <div
      className="reserved-tokens-item-container-inner"
    >
      <span
        className="reserved-tokens-item reserved-tokens-item-left"
      >
        0x22d491bde2303f2f43325b2108d26f1eaba1e32b
      </span>
      <span
        className="reserved-tokens-item reserved-tokens-item-middle"
      >
        tokens
      </span>
      <span
        className="reserved-tokens-item reserved-tokens-item-right"
      />
    </div>
    <div
      className="reserved-tokens-item-empty"
    >
      <a
        onClick={[Function]}
      >
        <span
          className="item-remove"
        />
      </a>
    </div>
  </div>
</div>
`;

exports[`tokens 1`] = `
<div
  className="reserved-tokens-container"
>
  <div
    className="reserved-tokens-input-container"
  >
    <div
      className="reserved-tokens-input-container-inner"
    >
      <div
        className="reserved-tokens-input-property reserved-tokens-input-property-left"
      >
        <label
          className="label"
        >
          Address
        </label>
        <input
          className="input"
          disabled={undefined}
          onBlur={undefined}
          onChange={[Function]}
          onKeyPress={undefined}
          onPaste={undefined}
          style={undefined}
          type="text"
          value=""
        />
        <p
          className="description"
        >
          Address where to send reserved tokens.
        </p>
        <p
          style={
            Object {
              "color": "red",
              "fontSize": "12px",
              "fontWeight": "bold",
              "height": "10px",
              "width": "100%",
            }
          }
        />
      </div>
      <div
        className="reserved-tokens-input-property reserved-tokens-input-property-middle"
      >
        <label
          className="label"
        >
          Dimension
        </label>
        <div
          className="radios-inline"
        >
          <label
            className="radio-inline"
          >
            <input
              checked={true}
              onChange={[Function]}
              type="radio"
              value="tokens"
            />
            <span
              className="title"
            >
              tokens
            </span>
          </label>
          <label
            className="radio-inline"
          >
            <input
              checked={false}
              onChange={[Function]}
              type="radio"
              value="percentage"
            />
            <span
              className="title"
            >
              percentage
            </span>
          </label>
        </div>
        <p
          className="description"
        >
          Fixed amount or % of crowdsaled tokens. Will be deposited to the account after finalization of the crowdsale.
        </p>
      </div>
      <div
        className="reserved-tokens-input-property reserved-tokens-input-property-right"
      >
        <label
          className="label"
        >
          Value
        </label>
        <input
          className="input"
          disabled={undefined}
          onBlur={undefined}
          onChange={[Function]}
          onKeyPress={[Function]}
          onPaste={[Function]}
          style={undefined}
          type="number"
          value=""
        />
        <p
          className="description"
        >
          Value in tokens. Don't forget to click + button for each reserved token.
        </p>
        <p
          style={
            Object {
              "color": "red",
              "fontSize": "12px",
              "fontWeight": "bold",
              "height": "10px",
              "width": "100%",
            }
          }
        />
      </div>
    </div>
    <div
      className="plus-button-container"
    >
      <div
        className="button button_fill button_fill_plus"
        onClick={[Function]}
      />
    </div>
  </div>
  <div
    className="reserved-tokens-item-container"
  >
    <div
      className="reserved-tokens-item-container-inner"
    >
      <span
        className="reserved-tokens-item reserved-tokens-item-left"
      >
        0x90f8bf6a479f320ead074411a4b0e7944ea8c9c1
      </span>
      <span
        className="reserved-tokens-item reserved-tokens-item-middle"
      >
        tokens
      </span>
      <span
        className="reserved-tokens-item reserved-tokens-item-right"
      />
    </div>
    <div
      className="reserved-tokens-item-empty"
    >
      <a
        onClick={[Function]}
      >
        <span
          className="item-remove"
        />
      </a>
    </div>
  </div>
  <div
    className="reserved-tokens-item-container"
  >
    <div
      className="reserved-tokens-item-container-inner"
    >
      <span
        className="reserved-tokens-item reserved-tokens-item-left"
      >
        0x90f8bf6a479f320ead074411a4b0e7944ea8c9c1
      </span>
      <span
        className="reserved-tokens-item reserved-tokens-item-middle"
      >
        percentage
      </span>
      <span
        className="reserved-tokens-item reserved-tokens-item-right"
      />
    </div>
    <div
      className="reserved-tokens-item-empty"
    >
      <a
        onClick={[Function]}
      >
        <span
          className="item-remove"
        />
      </a>
    </div>
  </div>
  <div
    className="reserved-tokens-item-container"
  >
    <div
      className="reserved-tokens-item-container-inner"
    >
      <span
        className="reserved-tokens-item reserved-tokens-item-left"
      >
        0xffcf8fdee72ac11b5c542428b35eef5769c409f0
      </span>
      <span
        className="reserved-tokens-item reserved-tokens-item-middle"
      >
        percentage
      </span>
      <span
        className="reserved-tokens-item reserved-tokens-item-right"
      />
    </div>
    <div
      className="reserved-tokens-item-empty"
    >
      <a
        onClick={[Function]}
      >
        <span
          className="item-remove"
        />
      </a>
    </div>
  </div>
  <div
    className="reserved-tokens-item-container"
  >
    <div
      className="reserved-tokens-item-container-inner"
    >
      <span
        className="reserved-tokens-item reserved-tokens-item-left"
      >
        0x22d491bde2303f2f43325b2108d26f1eaba1e32b
      </span>
      <span
        className="reserved-tokens-item reserved-tokens-item-middle"
      >
        tokens
      </span>
      <span
        className="reserved-tokens-item reserved-tokens-item-right"
      />
    </div>
    <div
      className="reserved-tokens-item-empty"
    >
      <a
        onClick={[Function]}
      >
        <span
          className="item-remove"
        />
      </a>
    </div>
  </div>
</div>
`;
